app/src/main/java/com/example/tasky/feature_authentication/domain/use_case/Register.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.tasky.feature_authentication.domain.use_case
+
+import com.example.tasky.feature_authentication.domain.repository.AuthRepository
+import com.example.tasky.feature_authentication.domain.validation.EmailMatcher
+
+class Register(
+    private val repository: AuthRepository,
+    private val matcher: EmailMatcher
+) {
+    suspend operator fun invoke(fullName: String, email: String, password: String) {
+        if(validateUserData(fullName, email, password)) {
+            repository.signUp(fullName, email, password)
+        }
```

Comment:
And what happens if it's invalid? How does the UI know which fields are invalid? This function would just silently do nothing and the caller assumes everything was fine